tck-test/server.cpp

```diff
@@ -119,9 +118,11 @@ class ServerResponder : public RSocketResponder {
   MarbleStore marbles_;
 };
 
-[[noreturn]] static void signal_handler(int signal) {
+std::promise<void> terminate;
+
+static void signal_handler(int signal) {
   LOG(INFO) << "Terminating program after receiving signal " << signal;
-  exit(signal);
+  terminate.set_value();
```

Comment:
What are the chances this is safe to do in a signal handler...  Regardless, I don't have a better solution as `SignalBarrier` isn't open source and it's very likely Linux-only code.

Any particular reason for `std::future` instead of `folly::Future`?